sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -80,6 +79,30 @@
 _SUCCEEDED = frozenset(["succeeded"])
 
 
+def update_api_version_of_status_link(
+    status_link: str, lro_options: Optional[Dict[str, Any]] = None
```

Comment:
Why are we accepting a `Dict` here? We are just going to ignore any keys we don't understand, no?

Reply:
Updated to accept api_version directly instead of lro_options